### PR TITLE
Add IBM MQ scaler e2e test

### LINKED ISSUE #1287: Add e2e test for IBM MQ Scaler
IBM MQ Scaler issue: https://github.com/kedacore/keda/issues/1253

### PR DESCRIPTION
**Pull request by ayoyu**
Description of what has been changed: 
- Add e2e tests for the IBM MQ scaler
- Add error handling when the `commandResponse[0].parameters` field is missing in case of an error as discussed in this comment https://github.com/kedacore/keda/issues/1287#issuecomment-2106297798 . Reference about the response body format (JSON schema): https://www.ibm.com/docs/en/ibm-mq/9.3?topic=adminactionqmgrqmgrnamemqsc-post-json-formatted-command#q133340___responseformat__title__1 
For example when querying the current queue depth of a non existing queue:
```
{
  "commandResponse": [{
    "completionCode": 2,
    "reasonCode": 2085,
    "message": ["AMQ8147E: IBM MQ object bar not found."]
  }],
  "overallReasonCode": 3008,
  "overallCompletionCode": 2
}
```
Example from the operator log with the new error handling:
```diff
2024-06-02T16:18:50Z	INFO	Detected resource targeted for scaling	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "ScaledObject": {"name":"ibmmq-consumer","namespace":"ibmmq-test-ns"}, "namespace": "ibmmq-test-ns", "name": "ibmmq-consumer", "reconcileID": "c064a17e-7cd1-4a1e-ba90-c3dc8f4b4a6d", "resource": "apps/v1.Deployment", "name": "ibmmq-consumer"}
+2024-06-02T16:18:51Z	ERROR	scale_handler	error getting scale decision	{"scaledObject.Namespace": "ibmmq-test-ns", "scaledObject.Name": "ibmmq-consumer", "scaler": "IBMMQScaler", "error": "error inspecting IBM MQ queue depth: failed to get the current queue depth parameter, reason: AMQ8147E: IBM MQ object FAKE.QUEUE not found."}
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).getScalerState
	/workspace/pkg/scaling/scale_handler.go:780
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).getScaledObjectState.func1
	/workspace/pkg/scaling/scale_handler.go:633
```

- Fix an embedded nil error by removing it

### Checklist
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #1287

### CODE REVIEW

**Review by zroubalik — COMMENTED** [review]
great, thanks a lot! Do you think you can also add a simple unit test for the fix you included?

**Comment by ayoyu** [comment]
> great, thanks a lot! Do you think you can also add a simple unit test for the fix you included?

Yes I will add it.

**Comment by ayoyu** [comment]
> great, thanks a lot! Do you think you can also add a simple unit test for the fix you included?

Test added for `getQueueDepthViaHTTP`

File: pkg/scalers/ibmmq_scaler_test.go [hunk 1]
```diff
@@ -128,3 +132,93 @@ func TestIBMMQGetMetricSpecForScaling(t *testing.T) {
...
+	},
+}
+
+func TestIBMMQScalerGetQueueDepthViaHTTP(t *testing.T) {
+	for _, testData := range testQueueDepthResults {
+		t.Run(testData.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+				writer.WriteHeader(testData.responseStatus)
+
+				if _, err := writer.Write([]byte(testData.bodyStr)); err != nil {
```
[the remark this answers is not in the document]
Comment (ayoyu) [reply pkg/scalers/ibmmq_scaler_test.go]: fixed

File: tests/scalers/ibmmq/ibmmq_test.go [hunk 2]
```diff
@@ -329,22 +330,41 @@ func installIbmmq(t *testing.T, kc *kubernetes.Clientset, data templateData) {
...
 			"--set queueManager.multiInstance=false "+
 			"--set queueManager.dev.secret.name=%s "+
 			"--set queueManager.dev.secret.adminPasswordKey=adminPassword "+
 			"--set queueManager.dev.secret.appPasswordKey=appPassword "+
-			"--namespace %s --wait",
-		ibmmqHelmChartReleaseName, queueManagerName, secretName, testNamespace))
+			"--namespace %s --wait --debug",
+		ibmmqHelmChartReleaseName, chartTempDir, queueManagerName, secretName, testNamespace))
+	// temp for debugging purpose
+	t.Log(string(todebug))
```
[the remark this answers is not in the document]
Comment (zroubalik) [reply tests/scalers/ibmmq/ibmmq_test.go:367]: yeah, sorry I completely missed that this PR is ready for another round of review. Feel free to ping me if there's another long delay :)

[the remark this answers is not in the document]
Comment (ayoyu) [reply tests/scalers/ibmmq/ibmmq_test.go:367]: Ok thanks :). Apparently the last e2e pipeline passes so I'm going to revert this change.

**Pull request merged**